src/services/Serialization.jl

```diff
@@ -248,24 +248,11 @@ function decodePackedType(dfg::AbstractDFG, varOrder::AbstractVector{Symbol}, ::
     return factordata
 end
 
-# TODO: REFACTOR THIS AS A JSON3 STRUCT DESERIALIZER.
 function fncStringToData(packtype::Type{<:AbstractPackedFactor}, data::String)
 
-    # Convert string to Named Tuples for kwargs
-    # packed_ = JSON3.read(data, GenericFunctionNodeData{packtype})
-    fncData = JSON3.read(data) #, NamedTuple) # Dict{String,Any})
-
-
-    # data isa AbstractString ? JSON2.read(data) : data
-    # JSON3.generate_type(packed.fnc)
-    # JSON3.generate_type(fncData["fnc"])
-    
-    # TODO use kwdef constructors instead,
-    # @info "user factor" fncData["fnc"]
-    restring = JSON3.write(fncData["fnc"])
-    # @show restring
-    # packT = JSON3.read(restring, packtype)
-    packT = packtype(;JSON2.read(restring)...)
+    # Read string as JSON object to use as kwargs
+    fncData = JSON3.read(data)
+    packT = packtype(;fncData.fnc...)
```

Comment:
Uhmm, this did not work for me when I tried this.  Maybe you fixed the issue somewhere else

Reply:
I made a change in iif also